static/app/views/metrics/metricQueryContextMenu.tsx

```diff
@@ -162,14 +160,17 @@ export function MetricQueryContextMenu({
               router
             );
           } else {
-            const metricsMeta = getVirtualMeta(metricsQuery.mri);
-            const targetProject = selectedProjects.find(
-              p => p.id === String(metricsMeta.projectIds[0])
-            );
-            navigateTo(
-              `/settings/projects/${targetProject?.slug || ':projectId'}/metrics/${formatMRI(metricsQuery.mri)}/edit/`,
-              router
-            );
+            const extractionRule = getExtractionRule(metricsQuery.mri);
+            if (extractionRule) {
+              openExtractionRuleEditModal(
+                {
+                  metricExtractionRule: extractionRule,
+                },
+                {
+                  onClose: refetch,
```

Comment:
Shouldn't we refetch only if the "Save" action happened, there is no need to refetch all the rules if user opened the modal and closed it without performing any action?

If this is too complicated to do, I understand why are we refetching `onClose` and we can keep it simple for now, but in that case I would still add a TODO comment

Reply:
true! ideally, I would update and invalidate queries. Though I am not sure how good we are with partially matching query keys in our react-query setup.
Will check in a follow-up 👍

Reply:
nvm, it actually was really easy -> done 👍